user-interface/src/main/java/life/qbic/datamanager/parser/MetadataConverter.java

```diff
@@ -0,0 +1,352 @@
+                  Collectors.toSet()));
+    }
+    var missingProperties = new ArrayList<>();
+    for (Entry<String, Integer> entry : hitMap.entrySet()) {
+      if (entry.getValue() == 0) {
+        missingProperties.add(entry.getKey());
+      }
+    }
+    if (missingProperties.isEmpty()) {
+      return true;
+    } else {
+      log.debug("Missing properties for proteomics measurement: %s".formatted(missingProperties));
+    }
+    return false;
+  }
+
+
+  enum ProteomicsMeasurementProperty {
+    MEASUREMENT_ID("measurement id"),
+    QBIC_SAMPLE_ID("qbic sample id"),
+    SAMPLE_POOL_GROUP("sample pool group"),
+    ORGANISATION_ID("organisation id"),
+    FACILITY("facility"),
+    INSTRUMENT("instrument"),
+    CYCLE("cycle/fraction name"),
+    DIGESTION_METHOD("digestion method"),
+    DIGESTION_ENZYME("digestion enzyme"),
+    ENRICHMENT_METHOD("enrichment method"),
+    INJECTION_VOLUME("injection volume (ul)"),
+    LC_COLUMN("lc column"),
+    LCMS_METHOD("lcms method"),
+    LABELING_TYPE("labeling type"),
+    LABEL("label"),
+    COMMENT("comment");
+
+    private final String name;
+
+    ProteomicsMeasurementProperty(String value) {
+      this.name = value;
+    }
+
+    static String trimValue(String value) {
+      return value.trim().toLowerCase();
+    }
+
+    static Optional<ProteomicsMeasurementProperty> fromString(String value) {
+      var trimmed = trimValue(value);
+      return Arrays.stream(ProteomicsMeasurementProperty.values())
+          .filter(property -> property.propertyName().equals(trimmed)).findFirst();
+    }
+
+    static boolean valueMatchesAnyProperty(String value) {
+      var trimmedValue = trimValue(value);
+      return Arrays.stream(MeasurementProperty.values()).map(MeasurementProperty::name)
+          .anyMatch(trimmedValue::equalsIgnoreCase);
+    }
+
+    public String propertyName() {
+      return name;
+    }
+
+  }
+
+  enum NGSMeasurementProperty {
+    MEASUREMENT_ID("measurement id"),
+    ORGANISATION_ID("organisation id"),
+    SAMPLE_POOL_GROUP("sample pool group"),
+    FACILITY("facility"),
+    INSTRUMENT("instrument"),
+    SEQUENCING_READ_TYPE("sequencing read type"),
+    LIBRARY_KIT("library kit"),
+    FLOW_CELL("flow cell"),
+    SEQUENCING_RUN_PROTOCOL("sequencing run protocol"),
+    INDEX_I7("index i7"),
+    INDEX_I5("index i5"),
+    COMMENT("comment");
+
+    private final String name;
+
+    NGSMeasurementProperty(String value) {
+      this.name = value;
+    }
+
+    /**
+     * Tries to convert an input property value to a known {@link NGSMeasurementProperty}.
+     * <p>
+     * Trailing whitespace will be ignored.
+     *
+     * @param value the presumed value to convert to a known {@link NGSMeasurementProperty}
+     * @return the matching property, or {@link Optional#empty()}.
+     * @since 1.4.0
+     */
+    static Optional<NGSMeasurementProperty> fromStringTrailingIgnored(String value) {
+      var trimmedValue = value.trim();
```

Comment:
`String:trim` will remove all trailing AND all leading whitespace. Is this intended here? From the method name I would expect the leading whitespace to not be removed and only the trailing to be handled.